discovery/discovery_test.go

```diff
@@ -265,10 +265,14 @@ func TestNewDBOrchestratorPoolCache_GivenListOfOrchs_CreatesPoolCacheCorrectly(t
 	orchs, err := node.Database.SelectOrchs(nil)
 	require.Nil(err)
 	assert.Len(orchs, 3)
+	orchsTest := make([]orchTest, 1)
```

Comment:
Instead of defining a new `orchTest` struct just for these tests could we:

- Accept `ethAddresses []ethcommon.Address` in `StubOrchestrators()`
- Loop through the members of `orchs` and let `db0` equal the next member in each iteration
- Check `assert.Contains(addresses, db0.ServiceURI)`
- Check `assert.Contains(ethAddresses, db0.EthereumAddr)`

Reply:
I think the benefit of having `orchTest` is that we can do an exact object comparison regardless of which we care about rather than comparing seperate fields. 

eg. In `TestNewOrchestratorPoolWithPred_TestPredicate` 
I want to make sure that an orchestrator with address X , ethAddress f(X) actually has a price setting of Y.

Reply:
> I think the benefit of having orchTest is that we can do an exact object comparison regardless of which we care about rather than comparing seperate fields.

👍 

> I want to make sure that an orchestrator with address X , ethAddress f(X) actually has a price setting of Y

I'm a bit confused by this statement - I don't see `orchTest` being used in `TestNewOrchestratorPoolWithPred_TestPredicate`.

Reply:
Right, bad example. I did slim down the usage of `orchTest` down in #1182 quite a bit so perhaps we can find a compromise there?